File: 1940-maximum-xor-for-each-query/1940-maximum-xor-for-each-query.cpp
class Solution {
public:
    vector<int> getMaximumXor(vector<int>& nums, int maximumBit) {
        int m = 0;
        vector<int> res;
        int n = (1 << maximumBit) - 1;
        for (int i : nums) {
            m ^= i;
            res.push_back(m ^ n);
        }
        reverse(res.begin(), res.end());
        return res;
    }
};

File: 1940-maximum-xor-for-each-query/1940-maximum-xor-for-each-query.py
class Solution:
    def getMaximumXor(self, nums: List[int], maximumBit: int) -> List[int]:
        m=0
        res=[]
        n=(2**maximumBit)-1
        for i in nums:
            m^=i
            k=m^(n)
            res.append(k)
        return res[::-1]

        